kernel/src/engine/parquet_stats_skipping/tests.rs

```diff
@@ -312,394 +136,71 @@ impl MinMaxTestFilter {
-        );
-
-        expect_eq!(
-            MinMaxTestFilter::new(LO.into(), HI.into()).apply_binary(
-                LessThan,
-                col,
-                &MID.into(),
-                inverted,
-            ),
-            Some(true), // min..max range includes both LT and GE
-            "{col} < {MID} (min: {LO}, max: {HI}, inverted: {inverted})"
-        );
-
-        expect_eq!(
-            MinMaxTestFilter::new(LO.into(), MID.into()).apply_binary(
-                LessThan,
-                col,
-                &HI.into(),
-                inverted,
-            ),
-            Some(!inverted),
-            "{col} < {HI} (min: {LO}, max: {MID}, inverted: {inverted})"
-        );
-
-        expect_eq!(
-            MinMaxTestFilter::new(MID.into(), HI.into()).apply_binary(
-                LessThan,
-                col,
-                &LO.into(),
-                inverted,
-            ),
-            Some(inverted),
-            "{col} < {LO} (min: {MID}, max: {HI}, inverted: {inverted})"
-        );
-
-        expect_eq!(
-            MinMaxTestFilter::new(MID.into(), MID.into()).apply_binary(
-                GreaterThanOrEqual,
-                col,
-                &MID.into(),
-                inverted,
-            ),
-            Some(!inverted),
-            "{col} >= {MID} (min: {MID}, max: {MID}, inverted: {inverted})"
-        );
-
-        expect_eq!(
-            MinMaxTestFilter::new(LO.into(), HI.into()).apply_binary(
-                GreaterThanOrEqual,
-                col,
-                &MID.into(),
-                inverted,
-            ),
-            Some(true), // min..max range includes both EQ and NE
-            "{col} >= {MID} (min: {LO}, max: {HI}, inverted: {inverted})"
-        );
-
-        expect_eq!(
-            MinMaxTestFilter::new(LO.into(), MID.into()).apply_binary(
-                GreaterThanOrEqual,
-                col,
-                &HI.into(),
-                inverted,
-            ),
-            Some(inverted),
-            "{col} >= {HI} (min: {LO}, max: {MID}, inverted: {inverted})"
-        );
-
-        expect_eq!(
-            MinMaxTestFilter::new(MID.into(), HI.into()).apply_binary(
-                GreaterThanOrEqual,
-                col,
-                &LO.into(),
-                inverted,
-            ),
-            Some(!inverted),
-            "{col} >= {LO} (min: {MID}, max: {HI}, inverted: {inverted})"
-        );
-    }
-}
 
 #[test]
-fn test_binary_le_gt() {
-    use BinaryOperator::*;
-
-    const LO: Scalar = Scalar::Long(1);
-    const MID: Scalar = Scalar::Long(10);
-    const HI: Scalar = Scalar::Long(100);
+fn test_eval_binary_comparisons() {
     let col = &column_expr!("x");
+    let five = &Scalar::from(5);
+    let ten = &Scalar::from(10);
+    let fifteen = &Scalar::from(15);
+    let null = &Scalar::Null(DataType::INTEGER);
+
+    let expressions = [
+        Expr::lt(col.clone(), ten.clone()),
```

Comment:
i guess doing this to share ref to the same scalar? otherwise could we do something like `5.into()`?

Reply:
It was mostly to avoid magic constants... otherwise `Expr::lt(column_name!("foo"), 5)` would work Just Fine. And maybe that's better?

Reply:
hm.. for simple numbers and esp in testing I would maybe advocate for the `Expr::lt(column_name!("foo"), 5)`? not a huge opinion :)

Reply:
I revisited this, and there are several challenges (tho I can still improve the code over what we have today):
1. Some operations do not take `impl Into<_>` and so passing a primitive literal like `1` does not compile.
2. Some operations need to take `Scalar` so they can sometimes pass NULL values
3. `ColumnName` is not `const` and so can't be made into a constant; and there are format strings relying on the current `col` that would be "not fun" to change.

Reply:
if we think this is worth tackling then yea maybe just make a follow-up issue to improve some of these ergonomics?

Reply:
I already made some improvements, hopefully it's enough for now.